lib/package_url/string_utils.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+class PackageURL
+  module StringUtils
+    private
+
+    def strip(string, char)
+      string.delete_prefix(char).delete_suffix(char)
+    end
+
+    def split_segments(string)
+      strip(string, '/').split('/')
+    end
+
+    def segment_present?(segment)
+      !segment.empty? && segment != '.' && segment != '..'
```

Comment:
I thought that the subpath and the namespace were parsed the same way. However, after studying https://github.com/package-url/purl-spec/blob/0b1559f76b79829e789c4f20e6d832c7314762c5/PURL-SPECIFICATION.rst#how-to-build-purl-string-from-its-components more carefully, it seems that the subpath should discard `.` and `..` but the namespace can contain them. Is this right?

Reply:
I've updated the PR to allow `.` and `..` to be present in the namespace, to maintain the existing behavior.